KomaMRIFiles/src/Sequence/WritePulseq.jl

```diff
@@ -0,0 +1,407 @@
+"""
+    typeon_obj = get_typeon_obj(seq::Sequence, type::String)
+
+Returns the vector `typeon_obj` of objects (RF, Grad or ADC) where the object is on in a
+given sequence `seq` according to the `type` ∈ ["rf", "gr", "adc"] argument.
+"""
+function get_typeon_obj(seq::Sequence, type::String)
+    type == "rf"  && return [s.RF[1] for s ∈ seq if is_RF_on(s)]
+    type == "gr"  && return [g for s ∈ seq for g ∈ s.GR[:,1] if sum(abs.(g.A)) > 0]
+    type == "adc" && return [s.ADC[1] for s ∈ seq if is_ADC_on(s)]
+    return []
+end
+
+"""
+    typeunique_obj_id = get_typeunique_obj_id(typeon_obj::Vector)
+
+Returns the vector `typeunique_obj_id` of objects (RF, Grad or ADC) which are unique given a
+input vector `typeon_obj` of objects which are on.
+"""
+function get_typeunique_obj_id(typeon_obj::Vector)
+    typeunique_obj_id = []
+    id_cnt = 1
+    for obj ∈ typeon_obj
+        if all([!(obj ≈ obj_unique) for (obj_unique, _) ∈ typeunique_obj_id])
+            push!(typeunique_obj_id, [obj, id_cnt])
+            id_cnt += 1
+        end
+    end
+    return typeunique_obj_id
+end
+
+"""
+    obj = get_obj(s::Sequence, type::String)
+
+Returns the object (RF, Grad or ADC) of a Sequence `s` (ideally a one-block sequence)
+according to the input `type` ∈ ["rf", "gx", "gy", "gz", "adc"]).
+"""
+function get_obj(s::Sequence, type::String)
+    type == "rf"  && return s.RF[1]
+    type == "gx"  && return s.GR[1,1]
+    type == "gy"  && return s.GR[2,1]
+    type == "gz"  && return s.GR[3,1]
+    type == "adc" && return s.ADC[1]
```

Comment:
Why are this functions not specialized depending on the type and use if statements?

Reply:
Removed in 2dd30b0